engine/api/workflow/dao_test.go

```diff
@@ -617,7 +617,12 @@ func TestUpdateSimpleWorkflowWithApplicationEnvPipelineParametersAndPayload(t *t
 	assert.Equal(t, app2.ID, w2.WorkflowData.Node.Context.ApplicationID)
 	assert.Equal(t, env.ID, w2.WorkflowData.Node.Context.EnvironmentID)
 
-	test.NoError(t, workflow.Delete(context.TODO(), db, cache, proj, w2))
+	tx, err := db.Begin()
+	defer tx.Rollback()
```

Comment:
could be better to put it after checking the error to not call Rollback on a nil transaction